styx-scheduler-service/src/main/java/com/spotify/styx/docker/KubernetesDockerRunner.java

```diff
@@ -145,23 +152,38 @@ public String start(WorkflowInstance workflowInstance, RunSpec runSpec) throws I
     }
   }
 
-  private void ensureSecret(WorkflowInstance workflowInstance, RunSpec runSpec) {
+  private void ensureSecrets(WorkflowInstance workflowInstance, RunSpec runSpec) {
+
     if (runSpec.serviceAccount().isPresent()) {
       final String secretName = buildSecretName(runSpec.serviceAccount().get());
       final Secret secret = client.secrets().withName(secretName).get();
       if (secret == null) {
         final String serviceAccount = runSpec.serviceAccount().get();
 
-        // TODO: create keys calling GCP API, base64 already
-        final String jsonKey = "";
-        final String p12Key = "";
+        // TODO: need gcp project of the service account
+        final String project = "TODO";
```

Comment:
I think it's required. It's part of the POST request url.

`projects/{PROJECT_ID}/serviceAccounts/{SERVICE_ACCOUNT_EMAIL}`